<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="ball" params="parent radius *origin">

    <link name="ball_link">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <sphere radius="${radius}"/>
        </geometry>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <sphere radius="${radius}"/>
        </geometry>
      </collision>
      <xacro:default_inertial mass="0.1"/>
    </link>

    <joint name="ball_joint" type="floating">
      <parent link="${parent}" />
      <child link = "ball_link" />
      <xacro:insert_block name="origin" />
    </joint>

    <gazebo reference="ball_link">
      <mu1>10</mu1>
      <mu2>10</mu2>
      <material>Gazebo/Red</material>
      <turnGravityOff>false</turnGravityOff>
    </gazebo>

  </xacro:macro>
</robot>
